### ISSUE TITLE
Create a new function to populate a new Target struct object

### ISSUE THREAD

**Issue by erleene** [sky-uk/osprey#41]
In reference to https://github.com/sky-uk/osprey/blob/master/client/target.go#L8

We want to have the ability to populate our Target structs fields in order for us to integrate the login strategy of osprey into our in-house tooling. 

The current upstream login implementation looks like so, 

```
targetData, err := retriever.RetrieveClusterDetailsAndAuthTokens(target)
            if err != nil {
                if state, ok := status.FromError(err); ok && state.Code() == codes.Unauthenticated {
                    log.Fatalf("Failed to log in to %s: %v", target.Name(), state.Message())
                }
                success = false
                log.Errorf("Failed to log in to %s: %v", target.Name(), err)
                continue
            }
            updateKubeconfig(target, targetData)
```

A way that we can integrate osprey's login strategy into our in-house tooling is by creating a new constructor of the Target object.

**Comment by davidmccormick** [comment]
Creating a public `Target` constructor sounds preferable as it would avoid exposing private implementation details.

**Comment by totahuanocotl** [comment]
@erleen this has been merged and released: [v2.1.0](https://github.com/sky-uk/osprey/releases/tag/v2.1.0)
